File: src/main/java/hu/elte/appdev/cicasite/model/entities/Report.java
package hu.elte.appdev.cicasite.model.entities;


import lombok.*;

import javax.persistence.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "REPORTS")
public class Report extends BaseEntity {

	@ManyToOne(targetEntity = User.class) private User user;

	@ManyToOne(targetEntity = User.class) private User reported_by;

	@Enumerated(EnumType.STRING) private ReportReason reason;

	public enum ReportReason {
		REASON1,
		REASON2,
		REASON3
	}
}

File: app-dev-frontend/src/app/model/Report.ts
import {User} from "../model/User";

export enum ReportReason {
  REASON1='REASON1', REASON2='REASON2', REASON3='REASON3'
}

export class Report  {
  reported_by: User;
  user: User;
  reason: ReportReason;


  constructor(reported_by: User, user: User, reason: ReportReason) {
    this.reported_by = reported_by;
    this.user = user;
    this.reason = reason;
  }
}
